src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,562 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+
+using System;
+using System.Device.I2c;
+using System.Threading.Tasks;
+
+namespace Iot.Device
+{
+    public class Bmp280 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private readonly CommunicationProtocol _communicationProtocol;
+        private bool _initialized = false;
+        private readonly Bmp280CalibrationData _calibrationData;
+        private const byte Signature = 0x58;
+        /// <summary>
+        /// The variable _temperatureFine carries a fine resolution temperature value over to the
+        /// pressure compensation formula and could be implemented as a global variable.
+        /// </summary>
+        private int _temperatureFine;
+
+        private enum CommunicationProtocol
+        {
+            I2c
+        }
+
+        public Bmp280(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+            _calibrationData = new Bmp280CalibrationData();
+            _communicationProtocol = CommunicationProtocol.I2c;
+        }
+
+        private void Begin()
+        {
+            _i2cDevice.WriteByte((byte)Registers.REGISTER_CHIPID);
+            var readSignature = _i2cDevice.ReadByte();
```

Comment:
In general, we try to avoid the use of `var` as much as possible since it might make the code less readable. One of the very few exceptions is when you are immediatly assigning it using the new keyword so the type is super obvious, like `var myType = new MyType();`. This feedback applies to the rest of your usage of `var`.